ovve_ui/utils/comms_link.py

```diff
@@ -316,6 +317,13 @@ def run(self) -> None:
             self.logger.debug('Serial Init Successful')
         else:
             self.logger.error('Serial Initialization failed')
-            # When the alarm infrastructure is done this would trigger an alarm
+            # Signal the UI to display and sound an alarm
+            self.lost_comms_signal.emit()
             return
-        self.process_SerialData()
+        
+        # Process serial data forever.  If we lose the connection,
+        # emit the the lost comms signal
+        try:
+            self.process_SerialData()
```

Comment:
`process_SerialData()` will not trigger an exception if it doesnt get anything for a while; so it doesnt really know the difference between receiving packets in a timely fashion and getting a single packet every 20 hours. Would probably need to modify this function to handle that. Will should at least trigger if the file handle for the serial port has a problem though, which is a comm error too.

Reply:
Ok, will check in this change for now and consider other ways to trigger the alarm.